crates/webidl/tests/all/consts.rs

```diff
@@ -0,0 +1,190 @@
+use super::project;
+
+#[test]
+fn bool() {
+    project()
+        .file(
+            "foo.webidl",
+            r#"
+                interface Foo {
+                    const boolean not_true = false;
+                    const boolean not_false = true;
+                };
+            "#,
+        )
+        // a corresponding const in the js implementation is not required
+        // value is taken directly from idl
+        .file(
+            "foo.js",
+            r#"
+                export class Foo {
+                }
+            "#,
+        )
+        .file(
+            "src/lib.rs",
+            r#"
+                #![feature(proc_macro, wasm_custom_section, wasm_import_module)]
+                extern crate wasm_bindgen;
+                use wasm_bindgen::prelude::*;
+
+                pub mod foo;
+                use foo::Foo;
+
+                #[wasm_bindgen]
+                pub fn test() {
+                    let falsish: bool = Foo::NOT_TRUE;
+                    assert!(!falsish);
+                    let trueish: bool = Foo::NOT_FALSE;
+                    assert!(trueish);
+                }
+            "#,
+        )
+        .test();
+}
+
+#[test]
+fn ints() {
+    project()
+        .file(
+            "foo.webidl",
+            r#"
+                interface Byte {
+                    const byte imin = -128;
+                    const byte imax = 127;
+                    const octet umin = 0;
+                    const octet umax = 255;
+                };
+                interface Short {
+                    const short imin = -32768;
+                    const short imax = 32767;
+                    const unsigned short umin = 0;
+                    const unsigned short umax = 65535;
+                };
+                interface Long {
+                    const long imin = -2147483648;
+                    const long imax = 2147483647;
+                    const unsigned long umin = 0;
+                    const unsigned long umax = 4294967295;
+                };
+                interface LongLong {
+                    const long long imin = -9223372036854775808;
+                    const long long imax = 9223372036854775807;
+                    const unsigned long long umin = 0;
+                    // bug in webidl
+                    // https://github.com/sgodwincs/webidl-rs/issues/15
+                    //const unsigned long long umax = 18446744073709551615;
```

Comment:
This is unfortunate. Upon scanning the docs, it doesn't look like [weedle](https://crates.io/crates/weedle) has this bug (it seems to preserve the literal string). I've wondered if we should use weedle instead since its dependencies are much more aligned with our own.

Reply:
It would also let us skip lalrpop, which is an incredible crate, but a long build time to hoist on to downstream users :-/

Reply:
Looks like it was already fixed, that was fast